Automattic/wp-calypso
# Add a wrapper for premium blocks

#### Changes proposed in this Pull Request

* This is a PoC branch to show how we might achieve a premium block flow.
* The basic concept is to wrap the block in a HOC which contains the Upgrade banner and a few other styles.
* Ideally we should be reusing the Calypso components rather than duplicating them, but we have to strike a balance between avoiding duplication, and shipping quickly.

#### Testing instructions
This is tricky to test
* Checkout this branch and build the app with `npx lerna run build --scope='@automattic/wpcom-block-editor' -- -- --output-path=[path to some dir to sync with your sandbox]
* Sync the changes with your sandbox
* Alternatively use the patch I created here: D29204-code (either way you'll need the other changes in D29204-code)
* Sandbox a wpcom simple site that doesn't have a plan
* Add a Simple Payments block to your site
* You should see the banner like this:

<img width="526" alt="Screenshot 2019-06-07 at 06 27 45" src="https://user-images.githubusercontent.com/275961/59083022-8d413b80-88ed-11e9-95ed-ed082060f1cf.png">
<img width="588" alt="Screenshot 2019-06-07 at 06 27 51" src="https://user-images.githubusercontent.com/275961/59083024-8d413b80-88ed-11e9-9c88-5293fee719d3.png">
